lib/rubocop/cop/rspec/stubbed_mock.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that message expectations are not combined with message stubs.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo).to receive(:bar).with(42).and_return("hello world")
+      #
+      #   # good (without spies)
+      #   allow(foo).to receive(:bar).with(42).and_return("hello world")
+      #   expect(foo).to receive(:bar).with(42)
+      #
+      class StubbedMock < Base
+        MSG = 'Do not stub your mock.'
```

Comment:
We need a better message here.

Reply:
'Do not stub methods of the mock object.'?

Reply:
I think many people don’t immediately know the differences between stubs and mocks, and wouldn’t understand where we are trying to nudge.

I think the message needs to explain that the cop is “offended” when people combine `expect` with `and_return`, and that they should use `allow` instead.

Reply:
What do you prefer:
 - "Do not use configure response with message expectations, use message allowances instead."
 - "Use message allowances instead of message expectations when configuring a response."
 - any other option?

Reply:
* "Prefer `allow` over `expect` when configuring a response". 
This one may need to be dynamic (as there are different methods for expectation); or it might be "Prefer allowance over expectation when configuring responses"

Reply:
> * "Prefer `allow` over `expect` when configuring a response".

I like the explicitness of that one, @Darhazer. 

I am not sure “allowance” can be used in this context. I think we need a native English speaker to help us out 😄

As for dynamic, do you mean `expect`, `is_expected`, `expect_any_instance_of` (`Language::Expectations::ALL`)? The current implementation only searches for `expect`…

Reply:
Nice catch on other expectation methods. 👍 
Loving the "Prefer ... to ..." wording as well.

Reply:
Wondering if should still highlight the stubbing part if we put the offence message this way:
```ruby
      is_expected.to receive(:bar).and_return(:baz)
                                  ^^^^^^^^^^^^^^^^^ Prefer `allow` to `is_expected` when configuring a response.
```
or should it become
```ruby
      is_expected.to receive(:bar).and_return(:baz)
      ^^^^^^^^^^^ Prefer `allow` to `is_expected` when configuring a response.
```

With `expect` and `expect_any_instance` it's clear. For a plain `allow`, should we advise `is_expected` as a replacement?

Reply:
`is_expected`'s replacement goes directly against `SubjectStub`. Going to recommend `allow(subject)` and let `SubjectStub` deal with it.